### PR TITLE
-Fix actors items

### PR DESCRIPTION
**Pull request by Shadowheg**
Bonjour, j'ai corrigé l'énumération des éléments dans les acteurs. Veuillez corriger d'autres champs personnalisés. J'aimerais pouvoir sélectionner des champs une seule fois, enregistrer les modifications en tant que modèle ou préréglage, et les utiliser lors de la récupération d'autres recueils.

### CODE REVIEW

File: scripts/exporters/actor-exporter.mjs [hunk 1]
```diff
@@ -22,32 +22,36 @@ export class ActorExporter extends AbstractExporter {
     const documents = await this.pack.getIndex({
       fields: [
         'prototypeToken.name',
+        'system.description.value', // Ensure you add fields relevant for your custom mapping here
```
Comment (DjLeChuck) [inline scripts/exporters/actor-exporter.mjs:25]: This is a system specific value, it shouldn’t be hardcoded. Please remove it.

File: scripts/exporters/abstract-exporter.mjs [hunk 2]
```diff
@@ -135,6 +135,21 @@ export class AbstractExporter {
...
+
+    // Serialize the transformed dataset to JSON
+    const jsonData = JSON.stringify(this.dataset, null, 2);
+
+    // Define the filename based on some unique attribute, like the pack's metadata ID
+    const filename = `${this.pack.metadata.id}-exported.json`;
+
+    // Use FileSaver (or Foundry's file handling API) to save the JSON data to a file
+    saveDataToFile(jsonData, 'text/json', `${this.pack.metadata.id}.json`); // Adjust this to match the actual API you're using
+  }
```
Comment (DjLeChuck) [inline scripts/exporters/abstract-exporter.mjs:151]: I don’t understand the aim of this change:
* If a piece of code is useless, remove it, no need to let it commented;
* You declare a `filename` variable but don’t use it;
* It seems you only move the code from `_getStringifiedDataset` to `_downloadFile`, am I right? Why this change?

File: scripts/exporters/actor-exporter.mjs [hunk 3]
```diff
@@ -2,17 +2,17 @@ import { AbstractExporter } from './abstract-exporter.mjs';
...
-      documentData.items = {};
-
-      for (const { name } of document.items) {
-        documentData.items[name] = { name };
-      }
-    }
-
+    const documentData = {
+      name: indexDocument.name,
+      description: document.system && document.system.description ? document.system.description.value : "", // Check for existence
```
Comment (DjLeChuck) [inline scripts/exporters/actor-exporter.mjs:7]: This is a system specific value, it shouldn’t be hardcoded. Please remove it.

File: scripts/exporters/actor-exporter.mjs [hunk 4]
```diff
@@ -2,17 +2,17 @@ import { AbstractExporter } from './abstract-exporter.mjs';
...
+      items: document.items.map(item => ({
+          id: item._id,
+          name: item.name,
+          description: item.system && item.system.description ? item.system.description.value : "" // Check for existence
```
Comment (DjLeChuck) [inline scripts/exporters/actor-exporter.mjs:11]: This is a system specific value, it shouldn’t be hardcoded. Please remove it.

File: scripts/exporters/actor-exporter.mjs [hunk 5]
```diff
@@ -22,32 +22,36 @@ export class ActorExporter extends AbstractExporter {
...
+            document,
+            this.options.customMapping.actor,
+        );
+
+        // Processing items with detailed information
+        if (document.items.size) {
+            documentData.items = document.items.map(item => ({
+                id: item._id,
+                name: item.name,
+                description: item.system.description.value // Ensure correct path based on your data structure
```
Comment (DjLeChuck) [inline scripts/exporters/actor-exporter.mjs:43]: This is a system specific value, it shouldn’t be hardcoded. Please remove it.

**Review by DjLeChuck — CHANGES_REQUESTED** [review]
Hi,

Thank you for your PR. I can’t accept it, please have a look at my comments.

**Comment by Shadowheg** [comment]
Hi. Thanks for the reply. I have a question then - which Json path should I specify so that it uploads the description of each item type object? in pf2e system. Like this 
"id": JSON.actors8].items[]._id
"description":JSON.actors[].items[].system.description.value

Implying that it will upload every description for every item for every actor. 

My fix allows me to get what I want.

**Pull request closed without merging**